os/fs/smartfs/smartfs_utils.c

```diff
@@ -1685,25 +1718,334 @@ struct smartfs_mountpt_s *smartfs_get_first_mount(void)
+					  */
+					ret = FS_IOCTL(fs, BIOC_FIBMAP, (unsigned long)firstsector);
+					if (ret < 0) {
+						fdbg("Get Bitmap Error ret : %d sector : %d\n", ret, firstsector);
+						goto errout;
+					}
+
+					/* It wasn't written, so clean current entry */
+					if ((ret == SMARTFS_ERASEDSTATE_16BIT) || (ret >= fs->fs_llformat.nsectors)) {
+						fdbg("Invalid entry, firstsector is not exist sector : %d firstsector : %d\n", logsector, firstsector);
+						ret = smartfs_invalidate_entry(fs, logsector, offset);
+						if (ret < 0) {
+							fdbg("Error marking entry inactive at sector %d\n", firstsector);
+							goto errout;
+						}
+
+					} else {
+						/* Make New node of Entry's first sector and put it to queue */
+						nodeitem = (struct sector_entry_queue_s *)kmm_malloc(sizeof(struct sector_entry_queue_s));
+						if (!node) {
+							fdbg("failed to alloc node.\n");
+							ret = -ENOMEM;
+							goto errout;
+						}
+						nodeitem->logsector = firstsector;
+						nodeitem->type = entrytype;
+						nodeitem->parentsector = logsector;
+						nodeitem->parentoffset = offset;
+#ifdef CONFIG_SMARTFS_ALIGNED_ACCESS
+						nodeitem->time = smartfs_rdle32(&entry->utc);
+#else
+						nodeitem->time = entry->utc;
+#endif
+						sq_addlast((FAR sq_entry_t *)nodeitem, &entry_queue);
+						fvdbg("new node added for currentsector : %d firsector : %d offset : %d\n",logsector, firstsector, offset);
+					}
+					offset += entrysize;
+
+				}
+			} 
+
+			/* Check next sector is valid or Not */
+			ret = FS_IOCTL(fs, BIOC_FIBMAP, (unsigned long)nextsector);
+			if (ret < 0) {
+				fdbg("Get Bitmap Error ret : %d sector : %d\n", ret, nextsector);
+				goto errout;
+			}
+			fvdbg("Nextsector : %d ret : 0x%x\n", nextsector, ret);
+
+			/* Previous Header was updated but Next sector wasn't written, clean chain */
+			if ((nextsector != SMARTFS_ERASEDSTATE_16BIT) && ((ret == SMARTFS_ERASEDSTATE_16BIT) || (ret >= fs->fs_llformat.nsectors))) {
+				fvdbg("next sector [%d] is not exist, reset sector [%d] header is_remain : %d\n", nextsector, logsector, IS_SECTOR_REMAIN(map, nextsector));
+				header->nextsector[0] = CONFIG_SMARTFS_ERASEDSTATE;
+				header->nextsector[1] = CONFIG_SMARTFS_ERASEDSTATE;
+				readwrite.logsector = logsector;
+				readwrite.offset = offsetof(struct smartfs_chain_header_s, nextsector);
+				readwrite.count = sizeof(uint16_t);
+				readwrite.buffer = header->nextsector;
+				nextsector = SMARTFS_ERASEDSTATE_16BIT;
+				ret = FS_IOCTL(fs, BIOC_WRITESECT, (unsigned long)&readwrite);
+				if (ret < 0) {
+					fdbg("Error unchaining sector (%d)\n", nextsector);
+					goto errout;
+				}
+
+				break;
+			}
+
+			logsector = nextsector;
+		}
+		
+		node = (struct sector_entry_queue_s *)sq_next(node);
+
+	}
+	ret = OK;
+errout :
+	node = (struct sector_entry_queue_s *)sq_peek(&entry_queue);
+	while (node != NULL) {
+		sq_rem((FAR sq_entry_t *)node, &entry_queue);
+		kmm_free(node);
+		node = (struct sector_entry_queue_s *)sq_next(node);
```

Comment:
After freeing node, and then you use it to get some data. It could cause system crash.